pytorch_lightning/strategies/ddp.py

```diff
@@ -257,6 +264,7 @@ def _reinit_optimizers_with_post_localSGD(self, warmup_steps: int):
             optimizers[x] = post_localSGD_optimizer
             del optimizer
         self.optimizers = optimizers
+        optimizers_to_device(self.optimizers, self.root_device)
```

Comment:
This shouldn't be needed with the proposed refactor above as this would be called in the parent setup class?